tortoise_interpreter.py

```diff
@@ -20,41 +20,54 @@ def sanitize_program(program):
 
 
 def navigate_command_functions(COMMANDS, program):
-    # Comment
-    return [f'{i}. {COMMANDS[command[0]](command)}' for i, command in enumerate(program, 1)]
+    command_to_parse = []
+    pen_state = ''
```

Comment:
So. I think that passing a state object from command to command is actually a really good idea :+1: However, why a string? If it was an object you'd be able to store more than one thing, give those items sensible names, etc etc.